schema/encoding/jsonschema/jsonschema.go

```diff
@@ -101,12 +101,20 @@ func validatorToJSONSchema(w io.Writer, v schema.FieldValidator) (err error) {
 		if t.ValuesValidator != nil {
 			ew.writeString(`, "items": `)
 			if ew.err == nil {
-				ew.err = validatorToJSONSchema(w, t.ValuesValidator)
+				ew.err = validatorToJSONSchema(w, t.ValuesValidator, true)
```

Comment:
While looking into that, I think I found a bug -- seams you put the "required" list inside "properties". This is incorrect, it should be outside (same level as "type": "object"). Otherwise it would mask out any field named "required".

See Basic Example:
- http://json-schema.org/examples.html

Sorry I missed this in the initial review!